crates/sui-types/src/effects/effects_v1.rs

```diff
@@ -129,24 +133,113 @@ impl TransactionEffectsAPI for TransactionEffectsV1 {
             })
             .collect()
     }
+
     fn created(&self) -> Vec<(ObjectRef, Owner)> {
         self.created.clone()
     }
+
     fn mutated(&self) -> Vec<(ObjectRef, Owner)> {
         self.mutated.clone()
     }
+
     fn unwrapped(&self) -> Vec<(ObjectRef, Owner)> {
         self.unwrapped.clone()
     }
+
     fn deleted(&self) -> Vec<ObjectRef> {
         self.deleted.clone()
     }
+
     fn unwrapped_then_deleted(&self) -> Vec<ObjectRef> {
         self.unwrapped_then_deleted.clone()
     }
+
     fn wrapped(&self) -> Vec<ObjectRef> {
         self.wrapped.clone()
     }
+
+    fn object_changes(&self) -> Vec<ObjectChange> {
+        let modified_at: BTreeMap<_, _> = self
+            .modified_at_versions
+            .iter()
+            .copied()
+            .collect();
+
+        let created = self
+            .created
+            .iter()
+            .map(|((id, v, _), _)| ObjectChange {
+                id: *id,
+                input: None,
+                output: Some(*v),
+                id_created: true,
+                id_deleted: false,
+            });
+
+        let mutated = self
+            .mutated
+            .iter()
+            .map(|((id, v, _), _)| ObjectChange {
+                id: *id,
+                input: modified_at.get(id).copied(),
```

Comment:
Flagging for review: If there's some issue/bug and an ID is missing from `modified_at_versions`, we'll just say that there is no input state (rather than error-ing out in some way).